### PR TITLE
expose sessionId and OnSessionEnter/Leave/Destroy

### PR DESCRIPTION
**Pull request by alouvion-aneo**
related issue : https://github.com/aneoconsulting/ArmoniK/issues/893
OnCreateService OnSessionEnter are not available in unifiedapi/sdk so we need to add this functionnality

### CODE REVIEW

File: Worker/src/Unified/GridWorker.cs [hunk 1]
```diff
@@ -223,16 +258,89 @@ public byte[] Execute(ITaskHandler taskHandler)
     return null;
   }
 
+
   public void SessionFinalize()
-  {
-  }
+    => OnSessionLeave();
```
Comment (lemaitre-aneo) [inline Worker/src/Unified/GridWorker.cs:263]: Shouldn't you have a check that Session has not already been closed?

File: Worker/src/Unified/ISubTaskingServiceConfiguration.cs [hunk 2]
```diff
@@ -28,18 +28,13 @@ namespace ArmoniK.DevelopmentKit.Worker.Unified;
 
 /// <summary>
 ///   implementation of this interface in <see cref="TaskWorkerService" /> or your own implementation allows to have the
-///   <see cref="ITaskHandler" /> configured automatically by the <see cref="GridWorker" />
+///   <see cref="SessionPollingService" /> configured automatically by the <see cref="GridWorker" />
 /// </summary>
-public interface ITaskWorkerServiceConfiguration
+internal interface ISessionServiceConfiguration
```
Comment (lemaitre-aneo) [inline Worker/src/Unified/ISubTaskingServiceConfiguration.cs:33]: Why is it internal?

File: Worker/src/Unified/GridWorker.cs [hunk 3]
```diff
@@ -223,16 +258,89 @@ public byte[] Execute(ITaskHandler taskHandler)
...
+
+
+    if (ServiceClass is ISessionConfiguration iSessionConfiguration)
+    {
+      if (iSessionConfiguration.SessionId == null || string.IsNullOrEmpty(iSessionConfiguration.SessionId.Id))
+      {
+        iSessionConfiguration.SessionId = session;
+      }
+
+      iSessionConfiguration.SessionId = session;
```
Comment (lemaitre-aneo) [inline Worker/src/Unified/GridWorker.cs:311]: Either the `if` is useless, or the assignment after the `if` should disappear.

Comment (alouvion-aneo) [reply Worker/src/Unified/GridWorker.cs:311]: indeed. updated

File: Worker/src/Unified/ISubTaskingServiceConfiguration.cs (hunk 2 above)
Comment (alouvion-aneo) [reply Worker/src/Unified/ISubTaskingServiceConfiguration.cs:33]: no reason. updated

**Review by ddubuc — APPROVED** [review]

**Pull request merged**